### PR TITLE
enh(ci): store docker slim images to github cache + workflow/action optimizations

### PR DESCRIPTION
**Pull request by kduret**
## Description

- store docker slim images to github cache
- upgrade dependencies (docker-compose, actions...)
- use properly pnpm cache
- do not install unnecessary binaries during pnpm install
- rename backend lint job (+ add it to package needs)
- improve cypress task to run docker container (more logs)
- remove `skip-grant-tables` from sql configuration of web container

**Fixes** MON-20421

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

### CODE REVIEW

File: .github/actions/frontend-build/action.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ inputs:
   base_directory:
     description: The base directory
     required: true
+  dependencies_lock_file:
+    description: "The frontend dependencies log file path"
```
Comment (sc979) [inline .github/actions/frontend-build/action.yml:8]:
```suggestion
    description: "The frontend dependencies lock file path"
```

File: .github/actions/frontend-lint/action.yml [hunk 2]
```diff
@@ -4,9 +4,9 @@ inputs:
   frontend_directory:
     description: "The frontend source directory"
     required: true
-  installation_directory:
-    description: "The installation directory"
-    required: false
+  dependencies_lock_file:
+    description: "The frontend dependencies log file path"
```
Comment (sc979) [inline .github/actions/frontend-lint/action.yml:8]:
```suggestion
    description: "The frontend dependencies lock file path"
```

File: .github/actions/lighthouse-performance-testing/action.yml [hunk 3]
```diff
@@ -16,6 +16,9 @@ inputs:
   path:
     description: "path"
     required: true
+  dependencies_lock_file:
+    description: "The frontend dependencies log file path"
```
Comment (sc979) [inline .github/actions/lighthouse-performance-testing/action.yml:20]:
```suggestion
    description: "The frontend dependencies lock file path"
```

File: .github/actions/lighthouse-performance-testing/action.yml [hunk 4]
```diff
@@ -24,6 +27,20 @@ runs:
         version: 7
         run_install: false
 
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 17
```
Comment (sc979) [inline .github/actions/lighthouse-performance-testing/action.yml:32]: not the same version between the actions ?
Here 17, frontend-lint in v18

File: .github/actions/web-frontend-component-test/action.yml [hunk 5]
```diff
@@ -7,6 +7,9 @@ inputs:
   spec_file_path:
     description: "Path of spec file."
     required: true
+  dependencies_lock_file:
+    description: "The frontend dependencies log file path"
```
Comment (sc979) [inline .github/actions/web-frontend-component-test/action.yml:11]:
```suggestion
    description: "The frontend dependencies lock file path"

```

File: .github/actions/cypress-e2e-testing/action.yml [hunk 6]
```diff
@@ -19,6 +19,9 @@ inputs:
   feature_file_path:
     description: "feature file"
     required: true
+  dependencies_lock_file:
+    description: "The frontend dependencies log file path"
```
Comment (Thebarda) [inline .github/actions/cypress-e2e-testing/action.yml]:
```suggestion
    description: "The frontend dependencies lock file path"
```

File: .github/actions/frontend-lint/action.yml (hunk 2 above)
Comment (Thebarda) [inline .github/actions/frontend-lint/action.yml]:
```suggestion
    description: "The frontend dependencies lock file path"
```

File: .github/actions/lighthouse-performance-testing/action.yml (hunk 3 above)
Comment (Thebarda) [inline .github/actions/lighthouse-performance-testing/action.yml]:
```suggestion
    description: "The frontend dependencies lock file path"
```

File: .github/actions/web-frontend-component-test/action.yml (hunk 5 above)
Comment (Thebarda) [inline .github/actions/web-frontend-component-test/action.yml]:
```suggestion
    description: "The frontend dependencies lock file path"

```

File: .github/workflows/cypress-component-parallelization.yml [hunk 7]
```diff
@@ -10,6 +10,10 @@ on:
       specs_path:
         required: true
         type: string
+      dependencies_lock_file:
+        required: true
+        type: string
+        description: "The frontend dependencies log file path"
```
Comment (sc979) [inline .github/workflows/cypress-component-parallelization.yml:16]:
```suggestion
        description: "The frontend dependencies lock file path"
```

Comment (Thebarda) [inline .github/workflows/cypress-component-parallelization.yml]:
```suggestion
        description: "The frontend dependencies lock file path"
```

File: .github/workflows/cypress-e2e-parallelization.yml [hunk 8]
```diff
@@ -25,6 +25,10 @@ on:
       package_directory:
         required: false
         type: string
+      dependencies_lock_file:
+        required: true
+        type: string
+        description: "The frontend dependencies log file path"
```
Comment (Thebarda) [inline .github/workflows/cypress-e2e-parallelization.yml]:
```suggestion
        description: "The frontend dependencies lock file path"
```

**Review by Thebarda — DISMISSED** [review]

File: .github/workflows/cypress-e2e-parallelization.yml [hunk 9]
```diff
@@ -82,6 +90,19 @@ jobs:
           username: ${{ secrets.registry_username }}
           password: ${{ secrets.registry_password }}
 
+      - name: Retore standard slim image in cache
```
Comment (sc979) [inline .github/workflows/cypress-e2e-parallelization.yml:93]:
```suggestion
      - name: Restore standard slim image in cache
```

File: .github/workflows/ui-beta.yml [hunk 10]
```diff
@@ -23,15 +23,16 @@ jobs:
       - uses: ./.github/actions/frontend-lint
         with:
           frontend_directory: ${{ env.directory }}
-          installation_directory: ${{ env.centreon_directory }}
           module_name: centreon-ui
+          dependencies_lock_file: centreon/pnpm-lock.yaml
```
Comment (sc979) [inline .github/workflows/ui-beta.yml:27]:
```suggestion
          dependencies_lock_file: ${{ env.centreon_directory }}/pnpm-lock.yaml
```
just a suggestion ;)

File: .github/workflows/ui-beta.yml [hunk 11]
```diff
@@ -23,15 +23,16 @@ jobs:
...
 
   cypress-component-testing:
     uses: ./.github/workflows/cypress-component-parallelization.yml
     with:
       name: component
       module_name: centreon/packages/ui
       specs_path: src/**
+      dependencies_lock_file: centreon/pnpm-lock.yaml
```
Comment (sc979) [inline .github/workflows/ui-beta.yml:35]:
```suggestion
      dependencies_lock_file: ${{ env.centreon_directory }}/pnpm-lock.yaml
```
same

**Review by Thebarda — DISMISSED** [review]

File: .github/workflows/web.yml [hunk 12]
```diff
@@ -100,11 +100,12 @@ jobs:
       - uses: ./.github/actions/frontend-build
         with:
           base_directory: ${{ env.base_directory }}
+          dependencies_lock_file: ${{ env.base_directory }}/pnpm-lock.yaml
           index_file: ${{ env.base_directory }}/www/index.html
           static_directory: ${{ env.base_directory }}/www/static
           index_cache_key: ${{ github.sha }}-${{ github.run_id }}-index
           static_cache_key: ${{ github.sha }}-${{ github.run_id }}-static
-          command: centreon:build
+          build_command: pnpm centreon:build
```
Comment (sc979) [inline .github/workflows/web.yml:108]: shouldn't you quote the command using ":" ?

Comment (kduret) [reply .github/workflows/web.yml:108]: not needed :)

**Review by bdauria — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**